File: back/DevC/src/main/java/com/back/devc/domain/interaction/postLike/controller/PostLikeController.java
package com.back.devc.domain.interaction.postLike.controller;

import com.back.devc.domain.interaction.postLike.dto.LikedPostResponse;
import com.back.devc.domain.interaction.postLike.dto.LikedPostsQuery;
import com.back.devc.domain.interaction.postLike.dto.PostLikeCommand;
import com.back.devc.domain.interaction.postLike.dto.PostLikeResponse;
import com.back.devc.domain.interaction.postLike.service.PostLikeService;
import com.back.devc.global.response.SuccessResponse;
import com.back.devc.global.response.successCode.PostLikeSuccessCode;
import com.back.devc.global.security.jwt.JwtPrincipal;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

import java.util.List;

import static com.back.devc.global.security.jwt.JwtPrincipalHelper.getAuthenticatedUserId;

@RestController
@RequestMapping("/api")
@RequiredArgsConstructor
public class PostLikeController {

    private final PostLikeService postLikeService;

    @PostMapping("/posts/{postId}/likes")
    public ResponseEntity<SuccessResponse<PostLikeResponse>> createLike(
            @AuthenticationPrincipal JwtPrincipal principal,
            @PathVariable Long postId
    ) {
        PostLikeCommand command = new PostLikeCommand(
                getAuthenticatedUserId(principal),
                postId
        );

        PostLikeResponse response = postLikeService.createLike(command);
        PostLikeSuccessCode successCode = PostLikeSuccessCode.POST_LIKE_CREATED;

        return ResponseEntity
                .status(successCode.getStatus())
                .body(SuccessResponse.of(successCode, response));
    }

    @DeleteMapping("/posts/{postId}/likes")
    public ResponseEntity<SuccessResponse<PostLikeResponse>> cancelLike(
            @AuthenticationPrincipal JwtPrincipal principal,
            @PathVariable Long postId
    ) {
        PostLikeCommand command = new PostLikeCommand(
                getAuthenticatedUserId(principal),
                postId
        );

        PostLikeResponse response = postLikeService.cancelLike(command);
        PostLikeSuccessCode successCode = PostLikeSuccessCode.POST_LIKE_CANCELED;

        return ResponseEntity
                .status(successCode.getStatus())
                .body(SuccessResponse.of(successCode, response));
    }

    @GetMapping("/users/me/likes")
    public ResponseEntity<SuccessResponse<List<LikedPostResponse>>> getLikedPosts(
            @AuthenticationPrincipal JwtPrincipal principal
    ) {
        LikedPostsQuery query = new LikedPostsQuery(
                getAuthenticatedUserId(principal)
        );

        List<LikedPostResponse> response = postLikeService.getLikedPosts(query);
        PostLikeSuccessCode successCode = PostLikeSuccessCode.LIKED_POSTS_FETCHED;

        return ResponseEntity
                .status(successCode.getStatus())
                .body(SuccessResponse.of(successCode, response));
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/interaction/postlike/controller/PostLikeController.kt
package com.back.devc.domain.interaction.postLike.controller

import com.back.devc.domain.interaction.postLike.dto.LikedPostResponse
import com.back.devc.domain.interaction.postLike.dto.LikedPostsQuery
import com.back.devc.domain.interaction.postLike.dto.PostLikeCommand
import com.back.devc.domain.interaction.postLike.dto.PostLikeResponse
import com.back.devc.domain.interaction.postLike.service.PostLikeService
import com.back.devc.global.response.SuccessResponse
import com.back.devc.global.response.successCode.PostLikeSuccessCode
import com.back.devc.global.security.jwt.JwtPrincipal
import com.back.devc.global.security.jwt.JwtPrincipalHelper.getAuthenticatedUserId
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api")
class PostLikeController(
    private val postLikeService: PostLikeService,
) {

    @PostMapping("/posts/{postId}/likes")
    fun createLike(
        @AuthenticationPrincipal principal: JwtPrincipal,
        @PathVariable postId: Long,
    ): ResponseEntity<SuccessResponse<PostLikeResponse>> {
        val command = PostLikeCommand(
            userId = getAuthenticatedUserId(principal),
            postId = postId,
        )

        val response = postLikeService.createLike(command)
        val successCode = PostLikeSuccessCode.POST_LIKE_CREATED

        return ResponseEntity
            .status(successCode.status)
            .body(SuccessResponse.of(successCode, response))
    }

    @DeleteMapping("/posts/{postId}/likes")
    fun cancelLike(
        @AuthenticationPrincipal principal: JwtPrincipal,
        @PathVariable postId: Long,
    ): ResponseEntity<SuccessResponse<PostLikeResponse>> {
        val command = PostLikeCommand(
            userId = getAuthenticatedUserId(principal),
            postId = postId,
        )

        val response = postLikeService.cancelLike(command)
        val successCode = PostLikeSuccessCode.POST_LIKE_CANCELED

        return ResponseEntity
            .status(successCode.status)
            .body(SuccessResponse.of(successCode, response))
    }

    @GetMapping("/users/me/likes")
    fun getLikedPosts(
        @AuthenticationPrincipal principal: JwtPrincipal,
    ): ResponseEntity<SuccessResponse<List<LikedPostResponse>>> {
        val query = LikedPostsQuery(
            userId = getAuthenticatedUserId(principal),
        )

        val response = postLikeService.getLikedPosts(query)
        val successCode = PostLikeSuccessCode.LIKED_POSTS_FETCHED

        return ResponseEntity
            .status(successCode.status)
            .body(SuccessResponse.of(successCode, response))
    }
}